packages/cpp/src/browser/cpp-task-provider.ts

```diff
@@ -37,10 +38,33 @@ export class CppTaskProvider implements TaskContribution, TaskProvider, TaskReso
     @inject(TaskResolverRegistry) protected readonly taskResolverRegistry: TaskResolverRegistry;
     @inject(TaskDefinitionRegistry) protected readonly taskDefinitionRegistry: TaskDefinitionRegistry;
     @inject(CppBuildConfigurationManager) protected readonly cppBuildConfigurationManager: CppBuildConfigurationManager;
+    @inject(ProblemMatcherRegistry) protected readonly problemMatcherRegistry: ProblemMatcherRegistry;
+    @inject(ProblemPatternRegistry) protected readonly problemPatternRegistry: ProblemPatternRegistry;
 
     @postConstruct()
     protected init(): void {
         this.registerTaskDefinition();
+        this.problemPatternRegistry.register({
+            'name': 'clangTidyPattern',
+            'regexp': '^(.+):(\\d+):(\\d+):\\s+(error|warning|info|note):\\s+(.+?)\\s+\\[(.+)\\]$',
+            'file': 1,
+            'line': 2,
+            'character': 3,
+            'severity': 4,
+            'message': 5,
+            'code': 6
+        });
+        this.problemMatcherRegistry.register({
+            'name': 'clangTidyMatcher',
+            'label': 'for clangd',
```

Comment:
Would a more appropriate label be `for clang-tidy`?

Reply:
Maybe following the same idea, the label `Clang-tidy problems` would be fine :)

Reply:
Clang-tidy problems
Done